docs/DeveloperGuide.md

```diff
@@ -463,16 +461,69 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
     Use case ends.
     
-**Use case : view modules**
+**Use case : UC04 View all current modules**
 
 **MSS**
 
-1. User requests to view current list of Modules
-2. GradPad shows current list of Modules
+1. User requests to view list of modules in `Current Modules`
+2. GradPad shows all modules added into `Current Modules`
+
+    Use case ends.
+    
+**User case: UC05 Edit a Module in `Current Modules`**
+
+1. User requests to list all modules in `Current Modules`
+2. GradPad shows the list of modules in `Current Modules`
+3. User requests to edit a module in `Current Modules`
+4. Module is replaced with updated fields
+    
+    Use case ends.
+
+**Extensions**
+
+* 2a. The list of modules in `Current Modules` is empty.
 
     Use case ends.
     
-**Use case : exit GradPad**
+* 3a. The given index is invalid.
+    
+    * 3a1. GradPad shows an error message.
+    
+        Use case resumes at step 2.
+
+**Use case : UC06 View required modules in syllabus**
+
+**MSS**
+
+1. User requests to view all required modules in syllabus
+2. GradPad displays the required modules in syllabus onto the `Command Line Display`
+
+    Use case ends.
+  
+**Use case : UC07 Search for module details**
+
+**MSS**
+
+1. User requests to search a module in the required module list.
```

Comment:
Actually it's not just the required module list righttt cos that's only like ~10 mods out of all 150 CS mods that they can search for

```suggestion
1. User requests to search for a module in the CS curriculum.
```